src/compiler/checker.ts

```diff
@@ -15064,7 +15064,10 @@ namespace ts {
 
         function getSuggestionForNonexistentSymbol(location: Node, name: __String, meaning: SymbolFlags): string {
             const result = resolveNameHelper(location, name, meaning, /*nameNotFoundMessage*/ undefined, name, /*isUse*/ false, (symbols, name, meaning) => {
-                // `name` from the callback === the outer `name`
+                // NOTE: `name` from the callback is supposed to  === the outer `name`, but is undefined in some cases
+                if (name === undefined) {

```

Comment:
would not it be better to put this in getActionsForCorrectSpelling? i find it hard to see how resolveName can return `undefined`